packages/adp-tooling/src/preview/client/api-handler.ts

```diff
@@ -0,0 +1,114 @@
+export const enum ApiEndpoints {
+    FRAGMENT = './adp/api/fragment',
+    CONTROLLER = './adp/api/controller',
+    MANIFEST_APP_DESCRIPTOR = '/adp/api/manifest'
+}
+
+export const enum RequestMethod {
+    GET = 'GET',
+    PUT = 'PUT',
+    POST = 'POST',
+    PATCH = 'PATCH',
+    DELETE = 'DELETE'
+}
+
+type Fragments = { fragmentName: string }[];
+
+export interface FragmentsResponse {
+    fragments: Fragments;
+    filteredFragments: Fragments;
+    message: string;
+}
+
+/**
+ * @description Class responsible for sending requests from client to the server
+ */
+export default class ApiRequestHandler {
```

Comment:
You don't need the overhead of a (UI5) class for a bunch of static functions, you can just write them as separate functions in a module:

```typescript
export async function request<T>(endpoint: ApiEndpoints, method: RequestMethod, data?: unknown): Promise<T> {
 ...
}

export async function getFragments<T>(): Promise<T> {
    return request<T>(ApiEndpoints.FRAGMENT, RequestMethod.GET);
}

...
```
You existing imports of the functionality should work as they are.

Reply:
Done in https://github.com/SAP/open-ux-tools/pull/1183/commits/8abb0baa0fe0466e9049545b5d940d72c6955ded